tools/osbuilder/initrd-builder/initrd_builder.sh

```diff
@@ -73,8 +73,9 @@ OK "init is installed"
 OK "Agent is installed"
 
 # initramfs expects /init, create symlink only if ${ROOTFS}/init does not exist
-# Init may be provided by other packages, e.g. systemd or GPU initrd/rootfs
-[ -x "${ROOTFS}/init" ] || [ -L "${ROOTFS}/init" ] || sudo ln -sf /sbin/init "${ROOTFS}/init"
+# ATTN: initrd /init cannot follow two levels of symlinks (/init to /sbin/init to /lib/systemd/systemd)
+# Setting /init directly to /lib/systemd/systemd as kata initiative to use systemd across project
+[ -x "${ROOTFS}/init" ] || [ -L "${ROOTFS}/init" ] || sudo ln -sf /lib/systemd/systemd "${ROOTFS}/init"
```

Comment:
@ryansavino, I think this is wrong.

If you do it here, you're breaking the agent as init case ... which is not used as part of CoCo anymore, but still used in the project.

If this is really necessary, I'd suggest:
* Expand this whole if / else
* Then add an if / else based on AGENT_INIT
  * There you can either link systemd or init accordingly

Does it make sense?

Reply:
Thanks for the tip. Fixed.